include/wil/registry_basic.h

```diff
@@ -172,7 +172,38 @@ namespace wil
 
         namespace reg_view_details
         {
+            template <typename T>
+            constexpr bool supports_prepare_buffer(T&) noexcept
+            {
+                return false;
+            }
+
+            template <typename T>
+            HRESULT prepare_buffer(T&) noexcept
+            {
+                // no-op in the default case
+                return S_OK;

```

Comment:
Shouldn't this throw `E_NOTIMPL` instead?

Reply:
No. we shouldn't fail the call. (this is really a no-op, but for compilers that don't support if constexpr, this function must be implemented anyway.)